Plugin crashes when trying to create custom phone provider

### Checklist

- [x] I have looked into the [README](https://github.com/auth0/terraform-provider-auth0#readme) and have not found a suitable solution or answer.
- [x] I have looked into the [documentation](https://registry.terraform.io/providers/auth0/auth0/latest/docs) and have not found a suitable solution or answer.
- [x] I have searched the [issues](https://github.com/auth0/terraform-provider-auth0/issues) and have not found a suitable solution or answer.
- [x] I have upgraded to the [latest version](https://github.com/auth0/terraform-provider-auth0/releases/latest) of this provider and the issue still persists.
- [x] I have searched the [Auth0 Community](https://community.auth0.com/c/sdks/5) forums and have not found a suitable solution or answer.
- [x] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

Below the complete error and the resource config.


Error in terminal:

```
terraform apply yeet
module.general.auth0_phone_provider.custom_phone_provider: Creating...
╷
│ Error: Plugin did not respond
│ 
│   with module.general.auth0_phone_provider.custom_phone_provider,
│   on general/custom_phone_provider.tf line 13, in resource "auth0_phone_provider" "custom_phone_provider":
│   13: resource "auth0_phone_provider" "custom_phone_provider" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-auth0_v1.20.1 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x104936d60]

goroutine 32 [running]:
github.com/auth0/terraform-provider-auth0/internal/auth0/branding.phoneProviderIsConfigured({0x104d80688?, 0x1400041dab0?}, 0x1400055aff8?)
        github.com/auth0/terraform-provider-auth0/internal/auth0/branding/resource_phone_provider.go:172 +0x40
github.com/auth0/terraform-provider-auth0/internal/auth0/branding.createPhoneProvider({0x104d80688, 0x1400041dab0}, 0x14000410900, {0x104c4fd80, 0x1400040cf60})
        github.com/auth0/terraform-provider-auth0/internal/auth0/branding/resource_phone_provider.go:120 +0x58
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x140000ed700, {0x104d805e0, 0x14000552cc0}, 0x14000410900, {0x104c4fd80, 0x1400040cf60})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/resource.go:838 +0xe4
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x140000ed700, {0x104d805e0, 0x14000552cc0}, 0x140002f09c0, 0x14000410780, {0x104c4fd80, 0x1400040cf60})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/resource.go:969 +0x874
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x14000609518, {0x104d805e0?, 0x14000552c00?}, 0x140003f2d70)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/grpc_provider.go:1188 +0xaa4
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x1400027d040, {0x104d805e0?, 0x14000552180?}, 0x1400041d420)
        github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/tf5server/server.go:866 +0x2a8
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x104d4ae60, 0x1400027d040}, {0x104d805e0, 0x14000552180}, 0x14000051c00, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:611 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x140000be000, {0x104d805e0, 0x14000552120}, 0x14000697140, 0x14000391c50, 0x1053ddb18, 0x0)
        google.golang.org/grpc@v1.69.4/server.go:1392 +0xc44
google.golang.org/grpc.(*Server).handleStream(0x140000be000, {0x104d80cd8, 0x1400043e1a0}, 0x14000697140)
        google.golang.org/grpc@v1.69.4/server.go:1802 +0x910
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.69.4/server.go:1030 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 27
        google.golang.org/grpc@v1.69.4/server.go:1041 +0x13c

Error: The terraform-provider-auth0_v1.20.1 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

Resource config:

```
resource "auth0_action" "send_custom_phone" {
  name    = "Custom Phone Provider"
  runtime = "node18"
  deploy  = true
  code    = file("${path.module}/files/custom-phone-provider.js")

  supported_triggers {
    id      = "custom-phone-provider"
    version = "v1"
  }
}

resource "auth0_phone_provider" "custom_phone_provider" {
  depends_on = [auth0_action.send_custom_phone] # Ensure the action is created first with `custom-phone-provider` as the supported_triggers
  name       = "custom"                         # Indicates a custom implementation
  disabled   = false                            # Disable the default phone provider
  configuration {
    delivery_methods = ["text"]
  }
  credentials {}
}
```

### Expectation

Plugin doesn't crash

### Reproduction

1. add new resource config to Terraform
2. run `terraform init`
3. run `terraform plan  -out yeet`
4. run `terraform apply yeet`
5. the request is cancelled because of ` runtime error: invalid memory address or nil pointer dereference`

### Auth0 Terraform Provider version

1.20.1

### Terraform version

v1.12.1 on darwin_arm64

--- a/internal/auth0/branding/resource_phone_provider.go
+++ b/internal/auth0/branding/resource_phone_provider.go
@@ -116,7 +116,7 @@ func NewPhoneProviderResource() *schema.Resource {
 func createPhoneProvider(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	api := meta.(*config.Config).GetAPI()
 
-	// Fetch list of Phone Providers
+	// Fetch list of Phone Providers.
 	phoneProviders, err := api.Branding.ListPhoneProviders(ctx)
 	if err != nil {
 		return diag.FromErr(internalError.HandleAPIError(data, err))

